homeassistant/components/lyric/sensor.py

```diff
@@ -43,210 +66,122 @@ async def async_setup_entry(
+                            value=lambda device: device.indoorTemperature,
+                        ),
+                        location,
+                        device,
+                    )
+                )
             if device.outdoorTemperature:
-                cls_list.append(LyricOutdoorTemperatureSensor)
+                entities.append(
+                    LyricSensor(
+                        coordinator,
+                        LyricSensorEntityDescription(
+                            key=f"{device.macID}_outdoor_temperature",
+                            name="Outdoor Temperature",
+                            device_class=DEVICE_CLASS_TEMPERATURE,
+                            state_class=STATE_CLASS_MEASUREMENT,
+                            native_unit_of_measurement=hass.config.units.temperature_unit,
+                            value=lambda device: device.outdoorTemperature,
+                        ),
+                        location,
+                        device,
+                    )
+                )
             if device.displayedOutdoorHumidity:
-                cls_list.append(LyricOutdoorHumiditySensor)
-            if device.changeableValues:
-                if device.changeableValues.nextPeriodTime:
-                    cls_list.append(LyricNextPeriodSensor)
-                if device.changeableValues.thermostatSetpointStatus:
-                    cls_list.append(LyricSetpointStatusSensor)
-            for cls in cls_list:
                 entities.append(
-                    cls(
+                    LyricSensor(
                         coordinator,
+                        LyricSensorEntityDescription(
+                            key=f"{device.macID}_outdoor_humidity",
+                            name="Outdoor Humidity",
+                            device_class=DEVICE_CLASS_HUMIDITY,
+                            state_class=STATE_CLASS_MEASUREMENT,
+                            native_unit_of_measurement="%",
+                            value=lambda device: device.displayedOutdoorHumidity,
+                        ),
                         location,
                         device,
-                        hass.config.units.temperature_unit,
                     )
                 )
+            if device.changeableValues:
+                if device.changeableValues.nextPeriodTime:
+                    entities.append(
+                        LyricSensor(
+                            coordinator,
+                            LyricSensorEntityDescription(
+                                key=f"{device.macID}_next_period_time",
+                                name="Next Period Time",
+                                device_class=DEVICE_CLASS_TIMESTAMP,
+                                value=lambda device: get_datetime_from_future_time(
+                                    device.changeableValues.nextPeriodTime
+                                ),
```

Comment:
This needs to return an isoformatted time. 
```suggestion
                                value=lambda device: get_datetime_from_future_time(
                                    device.changeableValues.nextPeriodTime
                                ).isoformat(),
```